lib/ecto/adapters/sql.ex

```diff
@@ -78,6 +78,18 @@ defmodule Ecto.Adapters.SQL do
       end
 
       @doc false
+      def stream(repo, meta, query, params, process, opts) do
+        opts =
```

Comment:
Given this `opts` update is the same as in `execute`, I would probably move it to Ecto.Adapters.SQL and make sure it is a private function that can be shared between `execute_and_cache`, `execute_or_reset` and `prepare_stream`. I would also use pattern matching on maps instead of `Map.fetch`, something like: `%{sources: tuple} when tuple_size(...)`.